device_backends/xdma/src/XdmaBackend.cc

```diff
@@ -108,8 +107,12 @@ namespace ChimeraTK {
   }
 
   std::string XdmaBackend::readDeviceInfo() {
-    // TODO implement me
-    return "";
+    if(!isOpen()) throw ChimeraTK::logic_error("Device not opened.");
```

Comment:
The device info usually is read before opening the device. I'll quickly change that to print 
`number of DMA channels = unkown (device closed)` instead of throwing.

Reply:
That's interesting, as I copied it from the [pcie backend implementation](https://github.com/ChimeraTK/DeviceAccess/blob/9e585a67a8222c856f1ef4aedceeda86b469df1f/device_backends/pcie/src/PcieBackend.cc#L277).

Reply:
Seems you found a bug in the PCIe backend. Plus it seems this function is not used very much, or at least not called on a closed device as I expected.